.gitignore

```diff
@@ -1,17 +1,83 @@
-*.iml
-*.ipr
+
+# Created by https://www.gitignore.io/api/java,jetbrains,gradle
```

Comment:
compared to previous ignores, it's (at least) missing eclipse AFAICT, not sure if other diffs still matter.